code/game/machinery/atmoalter/canister.dm

```diff
@@ -394,8 +407,10 @@ update_flag
 	data["menu"] = menu ? 1 : 0
 	data["canLabel"] = can_label ? 1 : 0
 	data["canister_color"] = canister_color
-	data["colorContainer"] = colorcontainer
-	data["possibleDecals"] = possibledecals
+	data["colorContainer"] = colorcontainer.Copy()
+	colorcontainer.Cut()
+	data["possibleDecals"] = possibledecals.Copy()
+	possibledecals.Cut()

```

Comment:
oh man it's a minor change in the middle of a tremendous code shift

well it looks alright, I don't see it possibly breaking things, since `colorcontainer` and `possibledecals` were reset in the old version of the proc anyways at lines 424-425